### PR TITLE
Add --test-command to bisect_packages.py, --fetch-latest-main to setup_venv.py.

### PR DESCRIPTION
**Pull request by ScottTodd**
Follow-up to https://github.com/iree-org/iree/pull/19289#discussion_r1866385869 and https://github.com/iree-org/iree/pull/19289#discussion_r1861273858.

### CODE REVIEW

File: build_tools/pkgci/setup_venv.py [hunk 1]
```diff
@@ -9,6 +9,28 @@
...
+  ```bash
+  python3.11 ./build_tools/pkgci/setup_venv.py /tmp/.venv --fetch-git-ref=main
+  ```
+
+* Locally, `--fetch-latest-main` can be used to download and setup the
+  venv from the latest completed run from the `main` branch in one step:
+
+  ```bash
+  python3.11 ./build_tools/pkgci/setup_venv.py /tmp/.venv --fetch-latest-main
+  ```
```
Comment (ScottTodd) [inline build_tools/pkgci/setup_venv.py:32]: Testing a bit to show how the downstream code here:
* https://github.com/nod-ai/SHARK-ModelDev/blob/d551ab1d0656831f945af7bafccdf80912d50615/.github/workflows/test_sdxl.yml#L50-L69
* https://github.com/nod-ai/SHARK-ModelDev/blob/main/ci-tools/latest-pkgci.py

can be updated:

```bash
cd ~/dev/scratch/iree/

# Create a fresh venv explicitly.
rm -rf 3.11.venv/
python3.11 -m venv 3.11.venv
source 3.11.venv/bin/activate

# Install some packages into the venv as usual.
pip install --index-url https://download.pytorch.org/whl/test/cpu torch==2.3.0

pip freeze
# filelock==3.13.1
# fsspec==2024.6.1
# Jinja2==3.1.4
# MarkupSafe==2.1.5
# mpmath==1.3.0
# networkx==3.3
# sympy==1.13.1
# torch==2.3.0+cpu
# typing_extensions==4.12.2

# Run setup_venv.py, pointing it at the existing venv.
python3.11 ./setup_venv.py ~/dev/scratch/iree/3.11.venv/ --fetch-latest-main
# Looking up latest workflow run for main branch
# Running command to find latest completed workflow run:
#   gh api -H Accept: application/vnd.github+json -H X-GitHub-Api-Version: 2022-11-28 /repos/iree-org/iree/actions/workflows/pkgci.yml/runs?branch=main&event=push&status=completed&per_page=1
# Found workflow run: https://github.com/iree-org/iree/actions/runs/12169259276
# Package iree-base-compiler not found in cache. Fetching from linux_x86_64_release_packages...
# Fetching artifacts for workflow run: 12169259276
# Found artifacts:
#   linux_x86_64_release_packages: /repos/iree-org/iree/actions/artifacts/2276241979/zip
# Downloading artifact /repos/iree-org/iree/actions/artifacts/2276241979/zip
# Extracting /tmp/tmpz09em92y/linux_x86_64_release_packages.zip
# Installing wheels: [(PosixPath('/tmp/tmpz09em92y'), 'iree-base-compiler'), (PosixPath('/tmp/tmpz09em92y'), 'iree-base-runtime')]
# Running command: /home/nod/dev/scratch/iree/3.11.venv/bin/python -m pip install --no-deps --no-index -f /tmp/tmpz09em92y --force-reinstall iree-base-compiler
# Looking in links: /tmp/tmpz09em92y
# Processing /tmp/tmpz09em92y/iree_base_compiler-3.1.0.dev0+df3491177531842799ea5dbfc52c591bcda56ffb-cp311-cp311-manylinux_2_27_x86_64.manylinux_2_28_x86_64.whl
# Installing collected packages: iree-base-compiler
# Successfully installed iree-base-compiler-3.1.0.dev0+df3491177531842799ea5dbfc52c591bcda56ffb
# Running command: /home/nod/dev/scratch/iree/3.11.venv/bin/python -m pip install --no-deps --no-index -f /tmp/tmpz09em92y --force-reinstall iree-base-runtime
# Looking in links: /tmp/tmpz09em92y
# Processing /tmp/tmpz09em92y/iree_base_runtime-3.1.0.dev0+df3491177531842799ea5dbfc52c591bcda56ffb-cp311-cp311-manylinux_2_28_x86_64.whl
# Installing collected packages: iree-base-runtime
# Successfully installed iree-base-runtime-3.1.0.dev0+df3491177531842799ea5dbfc52c591bcda56ffb

# Check that we have more packages installed, on top of the original venv.
pip freeze
# filelock==3.13.1
# fsspec==2024.6.1
# iree-base-compiler==3.1.0.dev0+df3491177531842799ea5dbfc52c591bcda56ffb
# iree-base-runtime==3.1.0.dev0+df3491177531842799ea5dbfc52c591bcda56ffb
# Jinja2==3.1.4
# MarkupSafe==2.1.5
# mpmath==1.3.0
# networkx==3.3
# sympy==1.13.1
# torch==2.3.0+cpu
# typing_extensions==4.12.2
```

File: build_tools/pkgci/setup_venv.py
Comment (ScottTodd) [inline build_tools/pkgci/setup_venv.py:1]: Could mention this script at https://iree.dev/reference/bindings/python/#prebuilt-packages :thinking: 

Like this:
image

Comment (ScottTodd) [reply build_tools/pkgci/setup_venv.py:1]: Stashing that for now: https://github.com/ScottTodd/iree/tree/docs-setup-venv

Comment (saienduri) [reply build_tools/pkgci/setup_venv.py:1]: Ooh this would be nice

**Review by saienduri — APPROVED** [review]

File: build_tools/pkgci/setup_venv.py
Comment (ScottTodd) [reply build_tools/pkgci/setup_venv.py:1]: Sent https://github.com/iree-org/iree/pull/19393 to mention on the website in a few places

**Pull request merged**